### PR TITLE
 lambda code modify.

### PR DESCRIPTION
**Pull request by yu199195**

### CODE REVIEW

File: shardingsphere-underlying/shardingsphere-common/src/main/java/org/apache/shardingsphere/underlying/common/database/type/DatabaseTypes.java [hunk 1]
```diff
@@ -40,8 +41,8 @@
     
     /**
      * Get name of trunk database type.
-     * 
      * @param databaseType database type
+     *
```
Comment (terrymanu) [inline shardingsphere-underlying/shardingsphere-common/src/main/java/org/apache/shardingsphere/underlying/common/database/type/DatabaseTypes.java]: Please do not modify original javadoc

File: shardingsphere-underlying/shardingsphere-common/src/main/java/org/apache/shardingsphere/underlying/common/database/type/DatabaseTypes.java [hunk 2]
```diff
@@ -50,8 +51,8 @@ public static String getTrunkDatabaseTypeName(final DatabaseType databaseType) {
     
     /**
      * Get trunk database type.
+     * @param name database name
```
Comment (terrymanu) [inline shardingsphere-underlying/shardingsphere-common/src/main/java/org/apache/shardingsphere/underlying/common/database/type/DatabaseTypes.java]: Please do not modify original javadoc

File: shardingsphere-underlying/shardingsphere-common/src/main/java/org/apache/shardingsphere/underlying/common/database/type/DatabaseTypes.java [hunk 3]
```diff
@@ -60,40 +61,29 @@ public static DatabaseType getTrunkDatabaseType(final String name) {
     
     /**
      * Get actual database type.
+     * @param name database name
```
Comment (terrymanu) [inline shardingsphere-underlying/shardingsphere-common/src/main/java/org/apache/shardingsphere/underlying/common/database/type/DatabaseTypes.java]: Please do not modify original javadoc

File: shardingsphere-underlying/shardingsphere-common/src/main/java/org/apache/shardingsphere/underlying/common/database/type/DatabaseTypes.java [hunk 4]
```diff
@@ -60,40 +61,29 @@ public static DatabaseType getTrunkDatabaseType(final String name) {
...
-        Preconditions.checkState(DATABASE_TYPES.containsKey(name), "Unsupported database: '%s'", name);
-        return DATABASE_TYPES.get(name);
+        return Optional.ofNullable(DATABASE_TYPES.get(name)).orElseThrow(() -> new ShardingSphereException("Unsupported database:'%s'", name));
     }
     
     /**
      * Get database type by URL.
-     * 
      * @param url database URL
+     *
```
Comment (terrymanu) [inline shardingsphere-underlying/shardingsphere-common/src/main/java/org/apache/shardingsphere/underlying/common/database/type/DatabaseTypes.java]: Please do not modify original javadoc

File: sharding-proxy/sharding-proxy-common/src/main/java/org/apache/shardingsphere/shardingproxy/config/ShardingConfigurationLoader.java [hunk 5]
```diff
@@ -64,10 +64,7 @@ public ShardingConfiguration load(final String path) throws IOException {
             });
         }
         Preconditions.checkState(!ruleConfigurations.isEmpty() || null != serverConfig.getOrchestration(), "Can not find any sharding rule configuration file in path `%s`.", configPath.getPath());
-        Map<String, YamlProxyRuleConfiguration> ruleConfigurationMap = new HashMap<>(ruleConfigurations.size(), 1);
-        for (YamlProxyRuleConfiguration each : ruleConfigurations) {
-            ruleConfigurationMap.put(each.getSchemaName(), each);
-        }
+        Map<String, YamlProxyRuleConfiguration> ruleConfigurationMap = ruleConfigurations.stream().collect(Collectors.toMap(YamlProxyRuleConfiguration::getSchemaName, each -> each));
         return new ShardingConfiguration(serverConfig, ruleConfigurationMap);
```
Comment (kimmking) [inline sharding-proxy/sharding-proxy-common/src/main/java/org/apache/shardingsphere/shardingproxy/config/ShardingConfigurationLoader.java]: can combine one line

**Pull request closed without merging**